public/app/features/alerting/unified/components/rule-editor/alert-rule-form/notificationMessageSectionExtensions/extensions.ts

```diff
@@ -0,0 +1,39 @@
+import { ComponentType, createElement } from 'react';
+
+import { t } from '@grafana/i18n';
+import { withErrorBoundary } from '@grafana/ui';
+
+import { logError } from '../../../../Analytics';
+
+export interface GenNotificationMessageSectionExtensionProps {}
+
+// Internal variable to store the extension component, for now only one component is supported
+let InternalNotificationMessageSectionExtension: ComponentType<GenNotificationMessageSectionExtensionProps> | null =
+  null;
+
+// This component is used to render the notification message section extension.
+export const NotificationMessageSectionExtension: ComponentType<GenNotificationMessageSectionExtensionProps> = (
+  props
+) => {
+  if (!InternalNotificationMessageSectionExtension) {
+    return null;
+  }
+
+  // Wrap the component with error boundary
+  const WrappedComponent = withErrorBoundary(InternalNotificationMessageSectionExtension, {
+    title: t(
+      'alerting.ai.error-boundary.notification-message-section-extension',
+      'Notification Message Section Extension failed to load'
+    ),
+    style: 'alertbox',
+    errorLogger: logError,
+  });
+
+  return createElement(WrappedComponent, props);
```

Comment:
Is the `createElement` function call necessary? Could we simplify to`<WrappedComponent {...props} />`?

Reply:
if I recall correctly, when using similar extension for ai buttons, we need this createElement, but i can doble-check it

Reply:
it worked! updated!